source/server/hot_restarting_base.cc

```diff
@@ -0,0 +1,213 @@
+#include "server/hot_restarting_base.h"
+
+#include "common/api/os_sys_calls_impl.h"
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Server {
+
+using HotRestartMessage = envoy::HotRestartMessage;
+
+void HotRestartingBase::initDomainSocketAddress(sockaddr_un* address) {
+  memset(address, 0, sizeof(*address));
+  address->sun_family = AF_UNIX;
+}
+
+sockaddr_un HotRestartingBase::createDomainSocketAddress(uint64_t id, const std::string& role) {
+  // Right now we only allow a maximum of 3 concurrent envoy processes to be running. When the third
+  // starts up it will kill the oldest parent.
+  const uint64_t MAX_CONCURRENT_PROCESSES = 3;
```

Comment:
MaxConcurrentProcesses per style guide preference, but actually I think we typically don't use special casing for local const, so max_concurrent_processes should be fine. We should probably clarify the style guide for class-member const and local const variables.

Also this can be a constexpr.

Reply:
This is pre-existing code. These seem like they belong in a followup; this PR is already plenty big.

Reply:
Agreed let's clean up in a follow up.